python/ray/dag/compiled_dag_node.py

```diff
@@ -1175,7 +1182,15 @@ def execute(
 
         self._get_or_compile()
 
-        inp = (args, kwargs)
+        if len(args) == 1 and len(kwargs) == 0:
+            # When serializing a tuple, the Ray serializer invokes pickle5, which adds
+            # several microseconds of overhead. One common case for accelerated DAGs is
+            # passing a single argument of type `bytes`. To avoid imposing this overhead
+            # on this common case, we create a fast path for this case that avoids
+            # pickle5.
+            inp = args[0]
+        else:
+            inp = (args, kwargs)
```

Comment:
Could we perhaps wrap this in a NamedTuple that we define internally? That way we can check `not isinstance(raw_args, RayDAGArgs)` instead of `isinstance(raw_args, bytes)` to improve the run time for all single-arg DAG executions, not just ones where the args is an instance of bytes.

Reply:
Done